src/xdg-autostart-generator/xdg-autostart-service.c

```diff
@@ -303,6 +306,72 @@ static int xdg_config_item_table_lookup(
         return config_item_table_lookup(table, section, lvalue, ret_func, ret_ltype, ret_data, userdata);
 }
 
+static int xdg_copy_systemd_service_sections(XdgAutostartService *service) {
+        int r;
+
+        _cleanup_fclose_ FILE *f = fopen(service->path, "re");
+        if (!f)
+                return log_error_errno(errno, "Failed to open configuration file '%s': %m", service->path);
+
+        _cleanup_free_ char *extra = NULL;
+        unsigned line = 0;
+        bool copy = false;
+        for (;;) {
+                _cleanup_free_ char *buf = NULL;
+                char *l;
+
+                r = read_line(f, LONG_LINE_MAX, &buf);
+                if (r == 0)
+                        break;
+                if (r == -ENOBUFS)
+                        return log_error_errno(r, "%s:%u: Line too long", service->path, line);
+                if (r < 0)
+                        return log_error_errno(r, "%s:%u: Error while reading configuration file: %m", service->path, line);
+
+                line++;
+
+                l = skip_leading_chars(buf, WHITESPACE);
+
+                if (l[0] != '[') {
+                        if (copy && !strextend(&extra, buf, "\n"))
+                                return log_oom();
+
+                        continue;
+                }
+
+                _cleanup_free_ char *n = NULL;
+                size_t k;
+
+                k = strlen(l);
+                assert(k > 0);
+
+                if (l[k-1] != ']')
```

Comment:
Does this mean that the parser will refuse a line like `[X-systemd Service] `? Shouldn't the whitespace stripping be done on both ends here?